### PR TITLE
Clean up sealed data by default

### PR DESCRIPTION
**Pull request by rranjan3**
Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

### CODE REVIEW

**Review by manju956 — APPROVED** [review]

File: docker/compose/avalon-multi-pool.yaml [hunk 1]
```diff
@@ -119,7 +119,7 @@ services:
       - 1948
     command: |
       bash -c "
-        if [ -v CLEAN_SEALED_DATA ] && [ $CLEAN_SEALED_DATA = 1 ]; then rm -f /project/avalon/.sealed/.sealed.data*; fi;
+        if [ ! -v CLEAN_SEALED_DATA ] || [ $CLEAN_SEALED_DATA = 1 ]; then rm -f /project/avalon/.sealed/.sealed.data*; fi;
```
Comment (Ram-srini) [inline docker/compose/avalon-multi-pool.yaml:122]: Add comment this conditional removing of file.

File: docs/FAQ.rst [hunk 2]
```diff
@@ -206,23 +206,21 @@ The fix is to remove ``/dev/sgx`` then install ``/dev/isgx``
...
-to ensure a smooth functioning in case of abrupt failures of the worker. Sealed data from
-respective worker enclaves are used to recover keys across restarts.
-However, this default behavior can be overriden if fresh pairs of aforementioned keys are
-desired. For this, it has to be ensured that the sealed data created during previous
-startup of the enclave manager (KME or Singleton) is cleaned up before starting again.
-This has been provisioned in the docker mode already. Prepend ``CLEAN_SEALED_DATA=1`` to
-the docker-compose command. For example -
+By default, workers use fresh set of signing and encryption key pairs (asymmetric keys) across
+restarts. Sealed data from respective worker enclaves are used to recover keys across restarts.
+However, the default behavior does not allow recovery from selaed data. In production, this behavior
```
Comment (manojgop) [inline docs/FAQ.rst:213]: selaed  -> sealed

File: docs/FAQ.rst [hunk 3]
```diff
@@ -206,23 +206,21 @@ The fix is to remove ``/dev/sgx`` then install ``/dev/isgx``
...
+can be overriden if old pairs of aforementioned keys are desired. For this, it has to be ensured that
```
Comment (manojgop) [inline docs/FAQ.rst:214]: overriden  -> overridden

File: docs/FAQ.rst (hunk 2 above)
Comment (manojgop) [inline docs/FAQ.rst:213]: We can remove "In production" since we can't assume which mode will be preferred in production

File: docs/FAQ.rst [hunk 4]
```diff
@@ -334,24 +332,26 @@ If using Docker, specify the URI as the name of the Docker container running
...
-change in the enclave itself across 2 lifetimes, with the second enclave trying to use
-the sealed data from the previous run causes this. The solution is to remove the sealed
-data before starting again. Refer to the steps mentioned in ``prevent a worker from
-using old keys`` in `Installation and Configuration`_ to achieve cleanup.
+some sealed data from respective enclave is used across enclave lifetimes now. The change
+in the enclave itself across two lifetimes, with the second enclave trying to use the sealed
+data from the previous run causes this. The way out of this is to remove the sealed data before
+starting again. Be cautious before you do this beacause this would result in creating fresh pair
+of signing and encryption keys. This would have further implications of inability to verify older
+work orders or ones pending to be processed.
```
Comment (manojgop) [inline docs/FAQ.rst:354]: This looks too verbose.  May be you can rephrase it differently to simplify it.

File: docker/compose/avalon-multi-pool.yaml (hunk 1 above)
Comment (rranjan3) [reply docker/compose/avalon-multi-pool.yaml]: Simplified the logic a bit Ram to make it understandable.

**Pull request merged**